### PR TITLE
fix(active-job): Honour dynamic changes in configuration

### PR DESCRIPTION
**Pull request by lavoiesl**
Fixes https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/1077 as introduced at https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/677#discussion_r1686598646

Added a unit test for good measure.

### CODE REVIEW

File: instrumentation/active_job/lib/opentelemetry/instrumentation/active_job/handlers/enqueue.rb [hunk 1]
```diff
@@ -28,10 +19,22 @@ def initialize(...)
...
             job = payload.fetch(:job)
-            span = tracer.start_span(@span_name_formatter.call(job), kind: :producer, attributes: @mapper.call(payload))
+            span = tracer.start_span(span_name(job), kind: :producer, attributes: @mapper.call(payload))
             OpenTelemetry.propagation.inject(job.__otel_headers) # This must be transmitted over the wire
             { span: span, ctx_token: OpenTelemetry::Context.attach(OpenTelemetry::Trace.context_with_span(span)) }
           end
+
+          private
+
+          def span_name(job)
```
Comment (lavoiesl) [inline instrumentation/active_job/lib/opentelemetry/instrumentation/active_job/handlers/enqueue.rb:29]: I think it'd be best to move this to the `Default` class and also change all the other spans to have a consistent naming, but I'm not sure if we want to lump this change in the PR, this was merely intended to fix a regression

**Comment by lavoiesl** [comment]
I don't understand the commit validation error:
```
  ❌ PullRequest: fix(active_job): Honour dynamic changes in configuration
  Error: PullRequest:1:4: error: A scope MAY be provided after a type. A scope MUST consist of a noun describing a section of the codebase surrounded by parenthesis, e.g., fix(parser):
  
    1 | fix(active_job): Honour dynamic changes in configuration
      |    ~~~~~~~~~~~~
```

It sounds like `fix(active_job):` should fit the bill 🤔

**Comment by kaylareopelle** [comment]
@lavoiesl - It looks like there are some Rubocop test failures preventing the 

> I don't understand the commit validation error:
> 
> ```
>   ❌ PullRequest: fix(active_job): Honour dynamic changes in configuration
>   Error: PullRequest:1:4: error: A scope MAY be provided after a type. A scope MUST consist of a noun describing a section of the codebase surrounded by parenthesis, e.g., fix(parser):
>   
>     1 | fix(active_job): Honour dynamic changes in configuration
>       |    ~~~~~~~~~~~~
> ```
> 
> It sounds like `fix(active_job):` should fit the bill 🤔

@lavoiesl  - I wonder if the underscore is the problem. Could you update to a hyphen?

**Comment by arielvalentin** [comment]
The tooling does support library scopes. Remove the scope `(active-job)` and it will pass.

**Comment by lavoiesl** [comment]
`fix(active-job)` works! But it's the title of the PR that needs to be changed, not the commit message 🤦🏼

**Comment by arielvalentin** [comment]
Yes sorry. It is indeed the title. When we merge PRs we squash all commits down into a single commit using the PR title.

**Comment by kaylareopelle** [comment]
@lavoiesl - Thanks for this PR! I'd like to get input from other members of the SIG about this change. I've added it to the agenda for tomorrow's meeting. 

You're welcome to join the SIG meeting if you're interested! You can access the calendar invite by [joining this Google Group.](https://groups.google.com/a/opentelemetry.io/g/calendar-ruby) We meet Tuesdays at 9:00 AM PT. 

Mostly, I'm concerned about setting a precedent to accommodate dynamic configuration. I'm not sure what OTel's stance is on this idea and want more input.

**Review by kaylareopelle — APPROVED** [review]
Hi @lavoiesl, we chatted about this in the SIG and are comfortable merging in this change. 

Though we don't condone nor support dynamic changes in configuration, we can make this adjustment to resolve the regression you're experiencing. 

Thank you for your submission and for adding tests too :)

**Comment by kaylareopelle** [comment]
@lavoiesl  - It looks like the branch is out-of-date with `main` and I don't have the permissions to update the branch for you. Could you update your branch so we can merge this in?

**Comment by lavoiesl** [comment]
Awesome, thank you! Updated 🚀

**Pull request merged**